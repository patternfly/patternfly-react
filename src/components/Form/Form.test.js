import React from 'react';
import { mount } from 'enzyme';

import {
  Button,
  DropdownButton,
  Icon,
  Form,
  Grid,
  MenuItem,
  Modal
} from '../../index';

test('Inline Form renders properly', () => {
  const component = mount(
    <Form inline>
      <Form.FormGroup controlId="email" placeholder="Email">
        <Form.FormControl type="email" placeholder="Email" />
      </Form.FormGroup>
      <Form.FormGroup controlId="password" placeholder="Password">
        <Form.FormControl type="password" placeholder="Password" />
      </Form.FormGroup>
      <Button bsStyle="primary">Log In</Button>
    </Form>
  );

  expect(component.render()).toMatchSnapshot();
});

test('Horizontal Form renders properly', () => {
  const component = mount(
    <Form horizontal>
      <Form.FormGroup controlId="name">
        <Grid.Col componentClass={Form.ControlLabel} sm={3}>
          Name
        </Grid.Col>
        <Grid.Col sm={9}>
          <Form.FormControl type="text" />
          <Form.HelpBlock>Enter your name</Form.HelpBlock>
        </Grid.Col>
      </Form.FormGroup>
      <Form.FormGroup controlId="address">
        <Grid.Col componentClass={Form.ControlLabel} sm={3}>
          Address
        </Grid.Col>
        <Grid.Col sm={9}>
          <Form.FormControl type="address" disabled />
          <Form.HelpBlock>Enter your address</Form.HelpBlock>
        </Grid.Col>
      </Form.FormGroup>
      <Form.FormGroup controlId="city">
        <Grid.Col componentClass={Form.ControlLabel} sm={3}>
          City
        </Grid.Col>
        <Grid.Col sm={9}>
          <Form.FormControl type="text" disabled />
          <Form.HelpBlock>Enter your city</Form.HelpBlock>
        </Grid.Col>
      </Form.FormGroup>
      <Form.FormGroup controlId="email">
        <Grid.Col componentClass={Form.ControlLabel} sm={3}>
          Email
        </Grid.Col>
        <Grid.Col sm={9}>
          <Form.FormControl type="email" disabled />
          <Form.HelpBlock>Enter a valid email address</Form.HelpBlock>
        </Grid.Col>
      </Form.FormGroup>
      <Form.FormGroup controlId="url">
        <Grid.Col componentClass={Form.ControlLabel} sm={3}>
          My meeting URL
        </Grid.Col>
        <Grid.Col sm={9}>
          <Form.InputGroup>
            <Form.FormControl type="url" />
            <Form.InputGroup.Button>
              <Button>Copy URL</Button>
            </Form.InputGroup.Button>
          </Form.InputGroup>
          <Form.HelpBlock>Enter a valid URL</Form.HelpBlock>
        </Grid.Col>
      </Form.FormGroup>
      <Grid.Row style={{ paddingTop: '10px', paddingBottom: '10px' }}>
        <Grid.Col smOffset={3} sm={9}>
          <span>
            <Button bsStyle="primary">Save</Button>
          </span>
          <span>
            <Button bsStyle="default">Cancel</Button>
          </span>
        </Grid.Col>
      </Grid.Row>
    </Form>
  );

  expect(component.render()).toMatchSnapshot();
});

test('Vertical Form renders properly', () => {
  const component = mount(
    <Form>
      <Grid.Row>
        <Grid.Col>
          <Form.FormGroup controlId="name">
            <Form.ControlLabel>Name</Form.ControlLabel>
            <Form.FormControl type="text" />
            <Form.HelpBlock>Enter your name</Form.HelpBlock>
          </Form.FormGroup>
          <Form.FormGroup controlId="address">
            <Form.ControlLabel>Address</Form.ControlLabel>
            <Form.FormControl type="address" />
            <Form.HelpBlock>Enter your address</Form.HelpBlock>
          </Form.FormGroup>
          <Form.FormGroup controlId="city">
            <Form.ControlLabel>City</Form.ControlLabel>
            <Form.FormControl type="text" />
            <Form.HelpBlock>Enter your city</Form.HelpBlock>
          </Form.FormGroup>
          <Form.FormGroup controlId="email">
            <Form.ControlLabel>Email</Form.ControlLabel>
            <Form.FormControl type="email" />
            <Form.HelpBlock>Enter a valid email address</Form.HelpBlock>
          </Form.FormGroup>
          <Form.FormGroup controlId="url">
            <Form.ControlLabel>My meeting URL</Form.ControlLabel>
            <Form.InputGroup>
              <Form.FormControl type="url" />
              <Form.InputGroup.Button>
                <Button>Copy URL</Button>
              </Form.InputGroup.Button>
            </Form.InputGroup>
            <Form.HelpBlock>Enter a valid URL</Form.HelpBlock>
          </Form.FormGroup>
        </Grid.Col>
      </Grid.Row>
      <Grid.Row style={{ paddingTop: '10px', paddingBottom: '10px' }}>
        <Grid.Col>
          <span>
            <Button bsStyle="primary">Save</Button>
          </span>
          <span>
            <Button bsStyle="default">Cancel</Button>
          </span>
        </Grid.Col>
      </Grid.Row>
    </Form>
  );

  expect(component.render()).toMatchSnapshot();
});

test('Modal Form renders properly', () => {
  const component = mount(
    <Modal>
      <Modal.Header>
        <Button className="close" aria-hidden="true" aria-label="Close">
          <Icon type="pf" name="close" />
        </Button>
        <Modal.Title>Basic Settings</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <Form horizontal>
          <Form.FormGroup controlId="name">
            <Grid.Col componentClass={Form.ControlLabel} sm={3}>
              Name
            </Grid.Col>
            <Grid.Col sm={9}>
              <Form.FormControl type="text" />
              <Form.HelpBlock>Enter your name</Form.HelpBlock>
            </Grid.Col>
          </Form.FormGroup>
          <Form.FormGroup controlId="address">
            <Grid.Col componentClass={Form.ControlLabel} sm={3}>
              Address
            </Grid.Col>
            <Grid.Col sm={9}>
              <Form.FormControl type="address" />
              <Form.HelpBlock>Enter your address</Form.HelpBlock>
            </Grid.Col>
          </Form.FormGroup>
          <Form.FormGroup controlId="city">
            <Grid.Col componentClass={Form.ControlLabel} sm={3}>
              City
            </Grid.Col>
            <Grid.Col sm={9}>
              <Form.FormControl type="text" />
              <Form.HelpBlock>Enter your city</Form.HelpBlock>
            </Grid.Col>
          </Form.FormGroup>
          <Form.FormGroup controlId="email">
            <Grid.Col componentClass={Form.ControlLabel} sm={3}>
              Email
            </Grid.Col>
            <Grid.Col sm={9}>
              <Form.FormControl type="email" />
              <Form.HelpBlock>Enter a valid email address</Form.HelpBlock>
            </Grid.Col>
          </Form.FormGroup>
          <Form.FormGroup controlId="url">
            <Grid.Col componentClass={Form.ControlLabel} sm={3}>
              My meeting URL
            </Grid.Col>
            <Grid.Col sm={9}>
              <Form.InputGroup>
                <Form.FormControl type="url" />
                <Form.InputGroup.Button>
                  <Button>Copy URL</Button>
                </Form.InputGroup.Button>
              </Form.InputGroup>
              <Form.HelpBlock>Enter a valid URL</Form.HelpBlock>
            </Grid.Col>
          </Form.FormGroup>
        </Form>
      </Modal.Body>
      <Modal.Footer>
        <Button bsStyle="default">Cancel</Button>
        <Button bsStyle="primary">Save</Button>
      </Modal.Footer>
    </Modal>
  );

  expect(component.render()).toMatchSnapshot();
});

test('Supported controls render properly', () => {
  const component = mount(
    <Form>
      <Form.FormGroup controlId="text">
        <Form.ControlLabel>Text</Form.ControlLabel>
        <Form.FormControl type="text" />
        <Form.HelpBlock>Help text</Form.HelpBlock>
      </Form.FormGroup>
      <Form.FormGroup controlId="email">
        <Form.ControlLabel>Email</Form.ControlLabel>
        <Form.FormControl type="email" />
        <Form.HelpBlock>Help text</Form.HelpBlock>
      </Form.FormGroup>
      <Form.FormGroup controlId="password">
        <Form.ControlLabel>Password</Form.ControlLabel>
        <Form.FormControl type="password" />
        <Form.HelpBlock>Help text</Form.HelpBlock>
      </Form.FormGroup>
      <Form.FormGroup controlId="file">
        <Form.ControlLabel>File</Form.ControlLabel>
        <Form.FormControl type="file" />
        <Form.HelpBlock>Help text</Form.HelpBlock>
      </Form.FormGroup>
      <Form.FormGroup controlId="checkbox">
        <Form.ControlLabel>Checkbox</Form.ControlLabel>
        <Form.Checkbox>Checkbox</Form.Checkbox>
        <Form.HelpBlock>Help text</Form.HelpBlock>
      </Form.FormGroup>
      <Form.FormGroup controlId="radio">
        <Form.ControlLabel>Radio</Form.ControlLabel>
        <Form.Radio>Radio</Form.Radio>
        <Form.HelpBlock>Help text</Form.HelpBlock>
      </Form.FormGroup>
      <Form.FormGroup controlId="checkboxGroup">
        <Form.ControlLabel>Checkbox Group</Form.ControlLabel>
        <div>
          <Form.Checkbox inline>1</Form.Checkbox>
          <Form.Checkbox inline>2</Form.Checkbox>
          <Form.Checkbox inline>3</Form.Checkbox>
        </div>
        <Form.HelpBlock>Help text</Form.HelpBlock>
      </Form.FormGroup>
      <Form.FormGroup controlId="radioGroup">
        <Form.ControlLabel>Radio Group</Form.ControlLabel>
        <div>
          <Form.Radio name="radioGroup" inline>
            1
          </Form.Radio>
          <Form.Radio name="radioGroup" inline>
            2
          </Form.Radio>
          <Form.Radio name="radioGroup" inline>
            3
          </Form.Radio>
        </div>
        <Form.HelpBlock>Help text</Form.HelpBlock>
      </Form.FormGroup>
      <Form.FormGroup controlId="textarea">
        <Form.ControlLabel>Textarea</Form.ControlLabel>
        <Form.FormControl componentClass="textarea" />
        <Form.HelpBlock>Help text</Form.HelpBlock>
      </Form.FormGroup>
      <Form.FormGroup controlId="static">
        <Form.ControlLabel>Static</Form.ControlLabel>
        <Form.FormControl.Static>email@example.com</Form.FormControl.Static>
        <Form.HelpBlock>Help text</Form.HelpBlock>
      </Form.FormGroup>
    </Form>
  );

  expect(component.render()).toMatchSnapshot();
});

test('Input Groups render properly', () => {
  const component = mount(
    <Form>
      <Grid.Row>
        <Grid.Col>
          <Form.FormGroup controlId="control-1">
            <Form.InputGroup>
              <Form.InputGroup.Addon>@</Form.InputGroup.Addon>
              <Form.FormControl type="text" />
            </Form.InputGroup>
          </Form.FormGroup>
          <Form.FormGroup controlId="control-2">
            <Form.InputGroup>
              <Form.FormControl type="text" />
              <Form.InputGroup.Addon>.00</Form.InputGroup.Addon>
            </Form.InputGroup>
          </Form.FormGroup>
          <Form.FormGroup controlId="control-3">
            <Form.InputGroup>
              <Form.InputGroup.Addon>$</Form.InputGroup.Addon>
              <Form.FormControl type="text" />
              <Form.InputGroup.Addon>.00</Form.InputGroup.Addon>
            </Form.InputGroup>
          </Form.FormGroup>
          <Form.FormGroup controlId="control-4">
            <Form.InputGroup>
              <Form.FormControl type="text" />
              <Form.InputGroup.Addon>
                <Icon name="music" />
              </Form.InputGroup.Addon>
            </Form.InputGroup>
          </Form.FormGroup>
          <Form.FormGroup controlId="control-5">
            <Form.InputGroup>
              <Form.InputGroup.Button>
                <Button>Before</Button>
              </Form.InputGroup.Button>
              <Form.FormControl type="text" />
            </Form.InputGroup>
          </Form.FormGroup>
          <Form.FormGroup controlId="control-6">
            <Form.InputGroup>
              <Form.FormControl type="text" />
              <DropdownButton
                componentClass={Form.InputGroup.Button}
                id="input-dropdown-addon"
                title="Action"
              >
                <MenuItem>Item</MenuItem>
              </DropdownButton>
            </Form.InputGroup>
          </Form.FormGroup>
          <Form.FormGroup controlId="control-7">
            <Form.InputGroup>
              <Form.InputGroup.Addon>
                <input type="radio" />
              </Form.InputGroup.Addon>
              <Form.FormControl type="text" />
            </Form.InputGroup>
          </Form.FormGroup>
          <Form.FormGroup controlId="control-8">
            <Form.InputGroup>
              <Form.InputGroup.Addon>
                <input type="checkbox" />
              </Form.InputGroup.Addon>
              <Form.FormControl type="text" />
            </Form.InputGroup>
          </Form.FormGroup>
        </Grid.Col>
      </Grid.Row>
    </Form>
  );

  expect(component.render()).toMatchSnapshot();
});
