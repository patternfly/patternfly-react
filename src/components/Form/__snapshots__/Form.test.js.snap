// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Horizontal Form renders properly 1`] = `
<form
  class="form-horizontal"
>
  <div
    class="form-group"
  >
    <label
      class="col-sm-3 control-label"
      for="name"
    >
      Name
    </label>
    <div
      class="col-sm-9"
    >
      <input
        class="form-control"
        id="name"
        type="text"
      />
      <span
        class="help-block"
      >
        Enter your name
      </span>
    </div>
  </div>
  <div
    class="form-group"
  >
    <label
      class="col-sm-3 control-label"
      for="address"
    >
      Address
    </label>
    <div
      class="col-sm-9"
    >
      <input
        class="form-control"
        disabled=""
        id="address"
        type="address"
      />
      <span
        class="help-block"
      >
        Enter your address
      </span>
    </div>
  </div>
  <div
    class="form-group"
  >
    <label
      class="col-sm-3 control-label"
      for="city"
    >
      City
    </label>
    <div
      class="col-sm-9"
    >
      <input
        class="form-control"
        disabled=""
        id="city"
        type="text"
      />
      <span
        class="help-block"
      >
        Enter your city
      </span>
    </div>
  </div>
  <div
    class="form-group"
  >
    <label
      class="col-sm-3 control-label"
      for="email"
    >
      Email
    </label>
    <div
      class="col-sm-9"
    >
      <input
        class="form-control"
        disabled=""
        id="email"
        type="email"
      />
      <span
        class="help-block"
      >
        Enter a valid email address
      </span>
    </div>
  </div>
  <div
    class="form-group"
  >
    <label
      class="col-sm-3 control-label"
      for="url"
    >
      My meeting URL
    </label>
    <div
      class="col-sm-9"
    >
      <span
        class="input-group"
      >
        <input
          class="form-control"
          id="url"
          type="url"
        />
        <span
          class="input-group-btn"
        >
          <button
            class="btn btn-default"
            type="button"
          >
            Copy URL
          </button>
        </span>
      </span>
      <span
        class="help-block"
      >
        Enter a valid URL
      </span>
    </div>
  </div>
  <div
    class="row"
    style="padding-top: 10px; padding-bottom: 10px;"
  >
    <div
      class="col-sm-9 col-sm-offset-3"
    >
      <span>
        <button
          class="btn btn-primary"
          type="button"
        >
          Save
        </button>
      </span>
      <span>
        <button
          class="btn btn-default"
          type="button"
        >
          Cancel
        </button>
      </span>
    </div>
  </div>
</form>
`;

exports[`Inline Form renders properly 1`] = `
<form
  class="form-inline"
>
  <div
    class="form-group"
    placeholder="Email"
  >
    <input
      class="form-control"
      id="email"
      placeholder="Email"
      type="email"
    />
  </div>
  <div
    class="form-group"
    placeholder="Password"
  >
    <input
      class="form-control"
      id="password"
      placeholder="Password"
      type="password"
    />
  </div>
  <button
    class="btn btn-primary"
    type="button"
  >
    Log In
  </button>
</form>
`;

exports[`Input Groups render properly 1`] = `
<form
  class=""
>
  <div
    class="row"
  >
    <div
      class=""
    >
      <div
        class="form-group"
      >
        <span
          class="input-group"
        >
          <span
            class="input-group-addon"
          >
            @
          </span>
          <input
            class="form-control"
            id="control-1"
            type="text"
          />
        </span>
      </div>
      <div
        class="form-group"
      >
        <span
          class="input-group"
        >
          <input
            class="form-control"
            id="control-2"
            type="text"
          />
          <span
            class="input-group-addon"
          >
            .00
          </span>
        </span>
      </div>
      <div
        class="form-group"
      >
        <span
          class="input-group"
        >
          <span
            class="input-group-addon"
          >
            $
          </span>
          <input
            class="form-control"
            id="control-3"
            type="text"
          />
          <span
            class="input-group-addon"
          >
            .00
          </span>
        </span>
      </div>
      <div
        class="form-group"
      >
        <span
          class="input-group"
        >
          <input
            class="form-control"
            id="control-4"
            type="text"
          />
          <span
            class="input-group-addon"
          >
            <span
              aria-hidden="true"
              class="fa fa-music"
            />
          </span>
        </span>
      </div>
      <div
        class="form-group"
      >
        <span
          class="input-group"
        >
          <span
            class="input-group-btn"
          >
            <button
              class="btn btn-default"
              type="button"
            >
              Before
            </button>
          </span>
          <input
            class="form-control"
            id="control-5"
            type="text"
          />
        </span>
      </div>
      <div
        class="form-group"
      >
        <span
          class="input-group"
        >
          <input
            class="form-control"
            id="control-6"
            type="text"
          />
          <span
            class="dropdown input-group-btn"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="dropdown-toggle btn btn-default"
              id="input-dropdown-addon"
              role="button"
              type="button"
            >
              Action 
              <span
                class="caret"
              />
            </button>
            <ul
              aria-labelledby="input-dropdown-addon"
              class="dropdown-menu"
              role="menu"
            >
              <li
                class=""
                role="presentation"
              >
                <a
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  Item
                </a>
              </li>
            </ul>
          </span>
        </span>
      </div>
      <div
        class="form-group"
      >
        <span
          class="input-group"
        >
          <span
            class="input-group-addon"
          >
            <input
              type="radio"
              value="on"
            />
          </span>
          <input
            class="form-control"
            id="control-7"
            type="text"
          />
        </span>
      </div>
      <div
        class="form-group"
      >
        <span
          class="input-group"
        >
          <span
            class="input-group-addon"
          >
            <input
              type="checkbox"
              value="on"
            />
          </span>
          <input
            class="form-control"
            id="control-8"
            type="text"
          />
        </span>
      </div>
    </div>
  </div>
</form>
`;

exports[`Modal Form renders properly 1`] = `null`;

exports[`Supported controls render properly 1`] = `
<form
  class=""
>
  <div
    class="form-group"
  >
    <label
      class="control-label"
      for="text"
    >
      Text
    </label>
    <input
      class="form-control"
      id="text"
      type="text"
    />
    <span
      class="help-block"
    >
      Help text
    </span>
  </div>
  <div
    class="form-group"
  >
    <label
      class="control-label"
      for="email"
    >
      Email
    </label>
    <input
      class="form-control"
      id="email"
      type="email"
    />
    <span
      class="help-block"
    >
      Help text
    </span>
  </div>
  <div
    class="form-group"
  >
    <label
      class="control-label"
      for="password"
    >
      Password
    </label>
    <input
      class="form-control"
      id="password"
      type="password"
    />
    <span
      class="help-block"
    >
      Help text
    </span>
  </div>
  <div
    class="form-group"
  >
    <label
      class="control-label"
      for="file"
    >
      File
    </label>
    <input
      class=""
      id="file"
      type="file"
    />
    <span
      class="help-block"
    >
      Help text
    </span>
  </div>
  <div
    class="form-group"
  >
    <label
      class="control-label"
      for="checkbox"
    >
      Checkbox
    </label>
    <div
      class="checkbox"
    >
      <label
        title=""
      >
        <input
          type="checkbox"
          value="on"
        />
        Checkbox
      </label>
    </div>
    <span
      class="help-block"
    >
      Help text
    </span>
  </div>
  <div
    class="form-group"
  >
    <label
      class="control-label"
      for="radio"
    >
      Radio
    </label>
    <div
      class="radio"
    >
      <label
        title=""
      >
        <input
          type="radio"
          value="on"
        />
        Radio
      </label>
    </div>
    <span
      class="help-block"
    >
      Help text
    </span>
  </div>
  <div
    class="form-group"
  >
    <label
      class="control-label"
      for="checkboxGroup"
    >
      Checkbox Group
    </label>
    <div>
      <label
        class="checkbox-inline"
        title=""
      >
        <input
          type="checkbox"
          value="on"
        />
        1
      </label>
      <label
        class="checkbox-inline"
        title=""
      >
        <input
          type="checkbox"
          value="on"
        />
        2
      </label>
      <label
        class="checkbox-inline"
        title=""
      >
        <input
          type="checkbox"
          value="on"
        />
        3
      </label>
    </div>
    <span
      class="help-block"
    >
      Help text
    </span>
  </div>
  <div
    class="form-group"
  >
    <label
      class="control-label"
      for="radioGroup"
    >
      Radio Group
    </label>
    <div>
      <label
        class="radio-inline"
        title=""
      >
        <input
          name="radioGroup"
          type="radio"
          value="on"
        />
        1
      </label>
      <label
        class="radio-inline"
        title=""
      >
        <input
          name="radioGroup"
          type="radio"
          value="on"
        />
        2
      </label>
      <label
        class="radio-inline"
        title=""
      >
        <input
          name="radioGroup"
          type="radio"
          value="on"
        />
        3
      </label>
    </div>
    <span
      class="help-block"
    >
      Help text
    </span>
  </div>
  <div
    class="form-group"
  >
    <label
      class="control-label"
      for="textarea"
    >
      Textarea
    </label>
    <textarea
      class="form-control"
      id="textarea"
    />
    <span
      class="help-block"
    >
      Help text
    </span>
  </div>
  <div
    class="form-group"
  >
    <label
      class="control-label"
      for="static"
    >
      Static
    </label>
    <p
      class="form-control-static"
    >
      email@example.com
    </p>
    <span
      class="help-block"
    >
      Help text
    </span>
  </div>
</form>
`;

exports[`Vertical Form renders properly 1`] = `
<form
  class=""
>
  <div
    class="row"
  >
    <div
      class=""
    >
      <div
        class="form-group"
      >
        <label
          class="control-label"
          for="name"
        >
          Name
        </label>
        <input
          class="form-control"
          id="name"
          type="text"
        />
        <span
          class="help-block"
        >
          Enter your name
        </span>
      </div>
      <div
        class="form-group"
      >
        <label
          class="control-label"
          for="address"
        >
          Address
        </label>
        <input
          class="form-control"
          id="address"
          type="address"
        />
        <span
          class="help-block"
        >
          Enter your address
        </span>
      </div>
      <div
        class="form-group"
      >
        <label
          class="control-label"
          for="city"
        >
          City
        </label>
        <input
          class="form-control"
          id="city"
          type="text"
        />
        <span
          class="help-block"
        >
          Enter your city
        </span>
      </div>
      <div
        class="form-group"
      >
        <label
          class="control-label"
          for="email"
        >
          Email
        </label>
        <input
          class="form-control"
          id="email"
          type="email"
        />
        <span
          class="help-block"
        >
          Enter a valid email address
        </span>
      </div>
      <div
        class="form-group"
      >
        <label
          class="control-label"
          for="url"
        >
          My meeting URL
        </label>
        <span
          class="input-group"
        >
          <input
            class="form-control"
            id="url"
            type="url"
          />
          <span
            class="input-group-btn"
          >
            <button
              class="btn btn-default"
              type="button"
            >
              Copy URL
            </button>
          </span>
        </span>
        <span
          class="help-block"
        >
          Enter a valid URL
        </span>
      </div>
    </div>
  </div>
  <div
    class="row"
    style="padding-top: 10px; padding-bottom: 10px;"
  >
    <div
      class=""
    >
      <span>
        <button
          class="btn btn-primary"
          type="button"
        >
          Save
        </button>
      </span>
      <span>
        <button
          class="btn btn-default"
          type="button"
        >
          Cancel
        </button>
      </span>
    </div>
  </div>
</form>
`;
